src/openforms/utils/json_logic/introspection.py

```diff
@@ -85,8 +85,12 @@
 }
 
 
-def introspect_json_logic(expression: JSON) -> ExpressionIntrospection:
-    tree = JSONLogicExpression.from_expression(expression).as_tree()
+def introspect_json_logic(expression: JSON) -> ExpressionIntrospection | str:
```

Comment:
I wouldn't do the exception handling here, but in a wrapper function that you define in `digest.py`

Converting exceptions into strings for code that is used in other places for other purposes seems like a bad idea - and a type checker would also start complaining here for all the other call-sites to `introspect_json_logic`!

As a rule of thumb - if you're modifying function signatures, you need to be damn sure it's really needed :)